cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -154,13 +201,25 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   private def readFrames(event: RecordedEvent): List[CallSite] = {
     val stack = event.getStackTrace
-
     if (stack eq null) List() else {
       stack.getFrames.iterator.asScala.map {
         frame: RecordedFrame =>
           val method = frame.getMethod
-          CallSite(method.getType.getName, method.getName, method.getName, method.getDescriptor, frame.getLineNumber)
-      }.distinct.takeWhile { f => !f.isIgnorableTopFrame }.toList
+          if(method ne null) {
+            val splitIndex = method.getType.getName.lastIndexOf(".");
+            CallSite(
+              method.getType.getName.substring(0,splitIndex).intern(),
+              method.getType.getName.substring(splitIndex+1).intern(),
+              method.getName.intern(),
+              method.getDescriptor.intern(),
+              frame.getLineNumber,
+              classLoaderKey
+            )
+          } else {
+            CallSite(new String("NoMethodInFrame_").intern, new String("NoMethodInFrame_").intern, new String("NoMethodInFrame_").intern, new String("NoMethodInFrame_").intern, frame.getLineNumber,classLoaderKey)
+          }
+      }.filter(_ ne null)
```

Comment:
how could it be null?

Reply:
That was part of the code we had already. I have checked that from different jfr files and not in any of them did I find a frame to be null. I does not have any effect when no frame is null so I left it there. maybe there is a situation where we will have a null frame coming in.

Reply:
what is the null check against?

Reply:
how could the list contain a null?

Reply:
```suggestion
      }
```

Reply:
> That was part of the code we had already. I have checked that from different jfr files and not in any of them did I find a frame to be null. I does not have any effect when no frame is null so I left it there. maybe there is a situation where we will have a null frame coming in.

you added this code. 

- Look at the PR
- Look at the code in the IDE  you are filtering the list that you just created a few lines above

Reply:
Oh yes, I added it. 
Initially when I found out the null methods in frames, I did not create a callsite, I just return a null value. and that filter was removing all those null callsites. It is not needed anymore.

Reply:
I have taken that out.